lib/tests/test.js

```diff
@@ -193,45 +179,45 @@ class Test {
 
   checkDeploymentOptions(options, callback) {
     const self = this;
-    if (!options.deployment) {
-      if (!self.simOptions.host && !self.simOptions.accounts) {
-        return callback();
-      }
-      self.simOptions = {};
+    let resetServices = false;
+    const {host, port, type, accounts} = options.deployment || {};
+
+    if (host && port && !['rpc', 'ws'].includes(type)) {
+      callback(__("contracts config error: unknown deployment type %s", type));
+    }
+
+    if(accounts || port !== this.simOptions.port || type !== this.simOptions.type || host !== this.simOptions.host) {
+      resetServices = true;
+    }
+    if (accounts) {
+      self.simOptions.accounts = AccountParser.parseAccountsConfig(accounts, self.web3);
     } else {
-      self.simOptions = {};
-      let resetServices = false;
-      if (options.deployment.accounts) {
-        // Account setup
-        self.simOptions.accounts = AccountParser.parseAccountsConfig(options.deployment.accounts, self.web3);
-        resetServices = true;
-      }
-      if (options.deployment.host && options.deployment.port && options.deployment.type) {
-        if (options.deployment.type !== 'rpc' && options.deployment.type !== 'ws') {
-          callback(__("contracts config error: unknown deployment type %s", options.deployment.type));
-        }
-        Object.assign(self.simOptions, {
-          host: options.deployment.host,
-          port: options.deployment.port,
-          type: options.deployment.type
-        });
-        resetServices = true;
-      }
-      if (!resetServices) {
-        return callback();
-      }
+      self.simOptions.account = null;
+    }
+
+    Object.assign(self.simOptions, {
+      host,
+      port,
+      type
+    });
+
+    if (!resetServices && !self.firstRunConfig) {
+      return callback();
     }
+
     self.initWeb3Provider((err) => {
       if (err) {
         return callback(err);
       }
+      self.firstRunConfig = false;
       self.initDeployServices();
       callback();
     });
   }
 
   config(options, callback) {
     const self = this;
+    self.needConfig = false;
```

Comment:
You probably need to set `self.needConfig` to true at the end of the config, so that a future test with no config still works.

Reply:
It is not needed because the connection is established so another test without config will be fine

Reply:
Ah ok, it's only for the connection. I thought it might have been for the requires too.